bugfix: eliminate runtime error in ciso diagnostics

klindsay28 opened the pull request:
remove store to unallocated field_3d components for vertical integrals
remove un-necessary store of zero to field_2d components for vertical integrals

passes aux_pop_MARBL test suite with gnu, with pop2/trunk_tags/cesm_pop_2_1_20160426

klindsay28 committed:
bugfix: eliminate runtime error in ciso diagnostics

remove store to unallocated field_3d components for vertical integrals
remove un-necessary store of zero to field_2d components for vertical integrals

mnlevy1981 commented:
Huh, I don't know how I got credit for adding the "In Progress" label... I tried to set up

https://waffle.io/NCAR/MARBL

And maybe that changed something?

Anyway, this addresses #40 and I'll bring it to master after resolving pull request #48

mnlevy1981 commented:
2 things:

1) I think the initialization in line 3469 is un-necessary as well
2) This fix is necessary (and sufficient) to run with NAG so GNU is not the only compiler that flagged this bug. I think we should add a CISO test for hobart_nag to the aux_pop_MARBL test suite as part of this pull request.

mnlevy1981 commented:
@I just realized that (2) above is actually in the POP tag, not MARBL... so it's just a one-line fix (removing line 3469 from `marbl_diagnostics_mod.F90`) in MARBL.

klindsay28 committed:
remove additional un-necessary stores

remove additional un-necessary store of zero to field_2d components for vertical integrals
